backend/instancemgmt/instance_manager.go

```diff
@@ -109,35 +128,28 @@ func (im *instanceManager) Get(ctx context.Context, pluginContext backend.Plugin
 		}
 	}
 
-	im.locker.Lock(cacheKey)
-	defer im.locker.Unlock(cacheKey)
+	im.locker.Lock(strKey)
+	defer im.locker.Unlock(strKey)
 
-	if item, ok := im.cache.Load(cacheKey); ok {
+	if item, ok := im.cache.Get(strKey); ok {
 		ci := item.(CachedInstance)
 		needsUpdate := im.provider.NeedsUpdate(ctx, pluginContext, ci)
 
 		if !needsUpdate {
 			return ci.instance, nil
 		}
 
-		if disposer, valid := ci.instance.(InstanceDisposer); valid {
-			time.AfterFunc(disposeTTL, func() {
-				disposer.Dispose()
-				activeInstances.Dec()
-			})
-		} else {
-			activeInstances.Dec()
-		}
+		im.cache.Delete(strKey)
 	}
 
 	instance, err := im.provider.NewInstance(ctx, pluginContext)
 	if err != nil {
 		return nil, err
 	}
-	im.cache.Store(cacheKey, CachedInstance{
+	im.cache.Set(strKey, CachedInstance{
```

Comment:
looks like there's a `cache.SetDefault()` [method](https://pkg.go.dev/github.com/patrickmn/go-cache#Cache.SetDefault) for the 'set this and use the default expiration' case.